src/parser/index.js

```diff
@@ -57,13 +58,57 @@ var parse = function(path){
         // not walked with the rest of the tree, but include things like "+" and
         // "-", which we need.
         node.terminalNodeText = [];
+
+        // Position information for the node
+        if (p === "enterMultiplicativeExpression"
+          || p === "enterAdditiveExpression"
+          || p === "enterTypeExpression"
+          || p === "enterUnionExpression"
+          || p === "enterInequalityExpression"
+          || p === "enterEqualityExpression"
+          || p === "enterMembershipExpression"
+          || p === "enterAndExpression"
+          || p === "enterOrExpression"
+          || p === "enterImpliesExpression"
+        ) {
+          const opSymbol = ctx.children[1].symbol;
+          node.start = { line: opSymbol.line, column: opSymbol.column };
+          node.length = opSymbol.stop - opSymbol.start + 1;
+          node.text = opSymbol.text;
+
+        } else if (p === "enterPolarityExpression") {
+          const pSymbol = ctx.children[0].symbol;
+          node.start = { line: pSymbol.line, column: pSymbol.column };
+          node.length = pSymbol.stop - pSymbol.start + 1;
+
+        } else if (p === "enterFunctionInvocation") {
+          const fSymbol = ctx.children[0].children[0];
+          node.start = { line: fSymbol.start.line, column: fSymbol.start.column };
+          node.text = fSymbol.getText();
+          node.length = node.text.length;
+
+        } else if (p === "enterMemberInvocation") {
+          const mSymbol = ctx.children[0].children[0].symbol;
+          node.start = { line: mSymbol.line, column: mSymbol.column };
+          node.length = mSymbol.stop - mSymbol.start + 1;
+          node.text = mSymbol.text;
+          node.length = node.text.length;
```

Comment:
node.length is being set twice to different things.

Reply:
I'll remove one of these.